### DISCUSSION TITLE
Love the full screen way of GUI

### DISCUSSION THREAD

**Discussion by defdefred** [VendelinSlezak/BleskOS#27] [Ideas]
What a great idea to ditch the windowed paradigm, thanks!

It seems that your OS is perfect to test drive different way of using computer and avoiding the more and more bloated software that make older hardware obsolete.

Could be an heaven for testing the slow web or alternative to the unicode and emoji cancer.

Do you think that porting existing application to BleskOS will be difficult?
I can think of several classical useful "small" softwares like:
- fim
- sc-im
- wordgrinder
- vim
- ssh client
- educational programming language

**Comment by VendelinSlezak** [comment]
Hello, thank you, I am glad that you like my idea :) I personally think that full screen GUI is really superior to classic windows GUI.

“Could be an heaven for testing the slow web or alternative to the unicode and emoji cancer.”
I am not exactly sure what do you mean by that, would you please explain this more?

“Do you think that porting existing application to BleskOS will be difficult?”
At this point of BleskOS code porting existing application would be at least as hard, as rewriting whole application, so I decided to take different approach. I want to create programs for BleskOS for all classic categories, and into those programs add all good ideas from other programs in that category. E. g. you will not need to decide between 1000 graphic programs, you will have just one that is perfectly debugged and integrated into BleskOS enviroment.

FIM - I must say I am not exactly sure what this program do, but if it is this project https://github.com/Achiefs/fim I think it can be integrated into File Manager.

SC-IM - This will be in Spreadsheet editor

WORDGRINDER/VIM - This could be implemented into Text editor

SSH CLIENT - At first we need drivers for secure internet connection, so after it we will be able to have something like this in BleskOS, probably as option in Internet Browser or maybe also File Manager.

EDUCATIONAL PROGRAMMING LANGUAGE - Interesting idea, maybe we can implement something like this.

You are more than welcome to make list of features you like in this programs and you would like to see in BleskOS, so then we can think about how to implement them.

Anyway, now I am taking some break from programming and I am focusing on other things in my life, but around new year I will be back and I am planning to add programs such as Document editor that will be able to work with ODT and DOCX and Spreadsheet editor that will be able to work with ODS and XLSX.

**Comment by defdefred** [comment]
FIM is a efficient image viewer: https://www.nongnu.org/fbi-improved/

> “Could be an heaven for testing the slow web or alternative to the unicode and emoji cancer.”
> I am not exactly sure what do you mean by that, would you please explain this more?
I have the feeling that computing is now over bloated to justify the ultra powerful computer available and exacerbating the programmed obsolescence.

An example is all the recent web site, running with much javascript and using lots of memory and cpu power (probably more to report personal data to the Gafam than making useful thing for the user).
The alternative could be a independent, more static, slow web limited to some markdown rendering capacity and avoiding the too cpu intensive https, but proposing an optional signature check when we choose to validate a document.

Another example, is the use of multibyte char instead of efficient 8bit character to code textual data. This use more memory for strings and font and more cpu to decode and display. And for what? displaying emoji, strange glyphes (some formating are now included in unicode!) and exotic character set we are never using.
The alternative I would like to see is a 8 bit char set for latin (probably the 7bit ascii standard) + the first bit giving another 7 bit of all possible diacritics we could add to the basic latin. Just like international US keyboard is working, with "^" + "i" = "î" or "," + "t" = "ţ".
So much possibility:
éèėêëěếềḗĕēẽęȩɇḕễḝẻȅȇểẹḙḛệ
ÉÈĖÊËĚẾỀḖĔĒẼĘȨɆḔỄḜẺȄȆỂẸḘḚỆ

Theses alternatives could help the use of older computer and recent but energy efficient one.

I'm not sure that mimic the modern internet is even possible for small computer...
